### ISSUE TITLE
How to mock dependencies? How to test Fragments/VM with Koin?

### ISSUE THREAD

**Issue by camino2007** [KasperskyLab/Kaspresso#218]
Hi, 

Sorry, maybe it's a dumb question and Kaspresso doesn't work the way I think it does. 
I really would like to give Kaspresso a try but I can' find any documentation for my problems. Let's say I have a fragment with a VM which contains a repository, injected by Koin. Actually a very common architecture. How can I now say to a mocked repository, return for different tests/use cases different values? 
Koin complains when I provide mocked objects like I did it in plain Espresso tests.

What I would like to do is something like this:
`@Test
    fun test() {
        before {
            activityTestRule.launchActivity(null)
        }.after {
            repository.addPages(listOf(Page(), Page()))
        }.run {
            step("Update page counter") {
                EditFragmentScreen {
                    viewPagerTextView.containsText("1 von 2")
                }
            }
        }
    }`

Something like this, doesn't work:
`class EditDocumentsFragmentTest2 : KoinTest, TestCase() {...}`

Long story short, my questions summarized:
1. How to provide mocked objects via Koin?
2. How to test a single fragment in a single-activity-multiple-fragment-setup? 
3. How to set  different return values for mocked objects?

Any help is welcome!

**Comment by matzuk** [comment]
Hi!
There is a [very good video](https://youtu.be/q_8UUhVDV7c?t=2426) explaining how to resolve such a kind of task. But this video in Russian. Maybe, English transcript will help you.
Also, there is a PR containing Best practices How to write autotests. This PR has a section devoted to tests with Mocks. PR - [here](https://github.com/KasperskyLab/Kaspresso/pull/162/files), see "## Advices". 
Apologies for not full information in English, but I hope it will help you.

**Comment by matzuk** [comment]
@camino2007 close the issue.
If you have questions be free to ask =)
